### PR TITLE
feat: create layout for swap and send

### PR DESCRIPTION
**Pull request by BZahory**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23741?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js [hunk 1]
```diff
@@ -24,6 +30,7 @@ export default class DomainInput extends Component {
 
   static propTypes = {
     className: PropTypes.string,
+    isBlockie: PropTypes.boolean,
```
Comment (micaelae) [inline ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js:33]: Nit: `useBlockie`

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js [hunk 2]
```diff
@@ -106,14 +114,19 @@ export default class DomainInput extends Component {
...
+                  variant={
+                    isBlockie
+                      ? AvatarAccountVariant.Blockies
+                      : AvatarAccountVariant.Jazzicon
+                  }
+                  address={selectedAddress}
+                  size={Size.SM}
+                  borderColor={BackgroundColor.backgroundDefault} // we currently don't have white color for border hence using backgroundDefault as the border
+                />
                 <div className="ens-input__selected-input__title">
```
Comment (micaelae) [inline ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js:127]: Correct me if I'm wrong, but I think we need to show `ENS ?? address` for the recipient, not the account name

Comment (micaelae) [reply ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js:127]: Screenshot 2024-04-08 at 3 35 19 PM

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js (hunk 1 above)
Comment (BZahory) [reply ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js]: Good call

File: ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js (hunk 2 above)
Comment (BZahory) [reply ui/pages/confirmations/send/send-content/add-recipient/domain-input.component.js:127]: Have this on the task list for the [revisions PR](https://github.com/MetaMask/metamask-extension/pull/23921)

**Comment by hmalik88** [comment]
hey folks, the snaps team has been updating the send flow to introduce domain resolution per snaps (already in flask, implementation is being updated to reflect changes in SIP-12). Just found out from @darkwing that this feature is being released in 11.16.0. This current PR is close to being merged this week: https://github.com/MetaMask/metamask-extension/pull/23697

File: ui/components/multichain/pages/send/send.js [hunk 3]
```diff
@@ -158,17 +199,29 @@ export const SendPage = () => {
...
+            // TODO: update to dest asset
+            amount={amount}
+            onAssetChange={handleSelectToken}
+            onAmountChange={(newAmount) =>
+              dispatch(updateSendAmount(newAmount))
             }
           />
-        ) : (
-          <SendPageRecipient />
         )}
```
Comment (micaelae) [inline ui/components/multichain/pages/send/send.js:212]: Is the `requireContractAddressAcknowledgement` logic not needed for the src token? I think it is in the current implementation

**Review by micaelae — APPROVED** [review]

**Pull request merged**